app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,67 @@
+require 'pry'
+
+class RentalsController < ApplicationController
+  def checkout
+    # take in query params
+    movie_id = params[:movie_id]
+    customer_id = params[:customer_id]
+
+    # find customer and movie by id's
+    movie = Movie.find_by(id: movie_id)
+    customer = Customer.find_by(id: customer_id)
+
+    checkout_date = Date.today
+    due_date = checkout_date + 7
+
+    # initialize a rental
+    rental = Rental.new(customer_id: customer_id, movie_id: movie_id, checkout_date: checkout_date, due_date: due_date)
+
+    if movie && customer
+
+      # save rental if valid, otherwise send errors
+      if rental.save
+
+        # increase the customer's checked-out movie number
+        customer.movies_checked_out_count += 1
+        customer.save
```

Comment:
You do a lot of work in this function: calculating dates, creating a `Rental`, increasing the movie count. It would be good to encapsulate all that in one model method.